compiler/front_end/glue.py

```diff
@@ -319,6 +319,7 @@ def process_ir(ir, stop_before_step):
         symbol_resolver.resolve_field_references,
         type_check.annotate_types,
         type_check.check_types,
+        constraints.check_early_constraints,
```

Comment:
Out of curiosity does this additional pass add much overhead to the end-to-end codegen time?

Reply:
```
> python3 -m cProfile ./embossc --output-path ~/scratch ~/scratch/large.emb

         25766374 function calls (23764959 primitive calls) in 17.613 seconds

   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
    181/1    0.014    0.000   18.280   18.280 {built-in method builtins.exec}
[...]
        1    0.000    0.000    0.003    0.003 constraints.py:717(check_early_constraints)
[...]
        8    0.000    0.000    0.002    0.000 constraints.py:365(_check_type_requirements_for_parameter_type)
[...]
```

Looks like <0.02% of the overall time compilation.  `large.emb` does have a few parameters, but is not parameter-heavy; I imagine that an `.emb` with a lot of parameters would pay a correspondingly higher compile time cost.